docs/rest-api-v1-v2-transition-guide.md

```diff
@@ -0,0 +1,116 @@
+# REST-API V1 V2 - Transition Guide
+
+## What Changed
+Assisted service is now using a V2 API, which introduces a new resource type called `infra_env`.
+
+`infra_env` represents the configuration needed to create the discovery ISO.
+Hosts will now register to `infra_env`, unlike in V1 API, where hosts registered directly to a `cluster`.
+This change comes to support more advanced use-cases, such as [late-binding](./hive-integration/late-binding.md).
```

Comment:
I think it's worth adding a couple of sentences about late binding here.
1. Hosts are first registered to an infra-env and then bound to a cluster at a later time.
2. Adding a cluster_id in the infra-env can cause hosts to automatically be bound upon registration.

Reply:
Done